### ISSUE TITLE
Connection established and shows tables, but fails when asking a question saying "can't find a schema for the database"

### ISSUE THREAD

**Issue by n00ge** [microsoft/copilot-pg#24]
I have a postgres connection established. 
When I run 
```
@pg /show

pg
Here are the tables in the database. You can ask for details about any table using show [table].
```
I get the output of my tables. 

However, when I subsequently run `@pg show all Users` (or really anything) I always get `I can't find the schema for the database. Please check the connection string with /conn.`. Note, if I re-run `@pg /show` that continues to work fine. It's not clear why `/show` works but asking it questions does not.

**Comment by robconery** [comment]
That's thrown when there's an error, and I should update that message. If you check the logs you should see the output message of the exact error. I'll add something more comprehensive in the next release - and thanks for letting me know!

**Comment by n00ge** [comment]
Apologies, but it's not clear what logs you'd be looking for. I'm looking in the Output panel when I run commands and am not seeing anything relevant. I've looked under "GitHub Copilot Chat", "GitHub Copilot Log", and "GitHub Copilot" in the dropdown of services. I'm happy to help if you can provide that information. Thanks!

**Comment by robconery** [comment]
Bah - OK thanks. Can you do me a favor and, using the same connection credentials, run the query:

```
select * from information_schema.tables;
```

This will help me rule out privilege stuff. I might not have actually deployed that fix yet.

**Comment by n00ge** [comment]
@robconery 
https://gist.github.com/n00ge/54da44a4da6e5e6b83b1c5a09dfe2223

Take a look here. Note I removed most of the tables but it should give you enough information. Let me know when you've reviewed. Thanks!

**Comment by robconery** [comment]
Thanks - how many tables do you have? I've been getting reports of failures on large table count.

**Comment by n00ge** [comment]
137 total "BASE TABLE" records in there.

**Comment by robconery** [comment]
Yeah, that's the issue. Copilot is bonking due to table size as it makes the prompt too big. Have to come up with a solution for that when I have a moment. Some way of telling it which tables to use for the prompt (Copilot needs to know the schema to answer your question).

**Comment by n00ge** [comment]
Very out of the weeds here but I'm curious. I imagine you're adding the full schema as context up front? Would it be feasible to just add the table names as context and then get the schema for the relevant tables in a follow up?

**Comment by robconery** [comment]
Good question - the problem is that I'm running the schema query first, without Copilot. I could _ask_ Copilot which tables it needs, I suppose, but then we get into a weird chicken/egg scenario. Not too sure what to do on this one. @isidorn you mention that we might have a larger request size for the API in the future - is that still the plan?

**Comment by isidorn** [comment]
@robconery yes, that is still the plan. We expect to increase the context size to 6K tokens soon, and then to 16K some time in the future (Sep+)

**Comment by enermark** [comment]
Same issue, 248 tables for the db Im testing against. Cool tool regardless.
